### PR TITLE
EXT: Resource Graph

### LINKED ISSUE #1736: Extensibility: Allow Resource Graphs in ext resources
**Description**
We want to allow Resource Graphs in Extensibility Details. Probably connections can be taken from `dataSources` from schema

**Acceptance criteria**
- [ ] Resource Graph is shown on resource details created by extensibility
- [ ] Resource Graph is showing connections

Proposed structure
```


dataSources: {
    Pizzas: {
        resource: {
            kind: "Pizza", required
            group: ...
            version: ...
            clusterWide: true, optional
        }
        filter: function/jsonata, required
    },
}


details:
    resourceGraph:
        depth: 2
        dataSources:
        -   source: Pizza
            other-shit
```

**Attachments**

### PR DESCRIPTION
**Pull request by Wawrzyn321**
**Description**

Changes proposed in this pull request:

- Add ResourceGraph handling to Extensibility
- Simplify our builtin config of relations
- Use per item comparison instead of list `$filter` in ext dataSources
- Fix APIRules graph not showing

**TODOs**

- optimize adding graph styles? Now multiple duplicate classes are added (which may not be a problem).

**Related issue(s)**

Closes #1736

### CODE REVIEW

**Review by wchrapka — APPROVED** [review]

File: docs/extensibility/resources.md [hunk 1]
```diff
@@ -133,7 +133,7 @@ The **list** section defines extra columns available in the list.
 
 ## _details_ section
 
-The **details** section defines the display structure for the details page. It contains two sections, `header` and `body`, both of which are a list of items to display in the **header** section and the body of the page respectively. The format of the entries is similar to the **form** section, however it has extra options available.
+The **details** section defines the display structure for the details page. It contains three sections, `header`, `body` and optional `resourceGraph`. First two sections are a list of items to display in the **header** section and the body of the page respectively. The format of the entries is similar to the **form** section, however it has extra options available. The `resourceGraph` section is used to configure the ResourceGraph which shows relationships between various resources.
```
Comment (grego952) [inline docs/extensibility/resources.md:136]:
```suggestion
The **details** section defines the display structure for the details page. It contains three sections, `header`, `body`, and optional `resourceGraph`. The first two sections are a list of items to display in the **header** section and the body of the page respectively. The format of the entries is similar to the **form** section, however it has extra options available. The `resourceGraph` section is used to configure the ResourceGraph which shows relationships between various resources.
```

File: docs/extensibility/resources.md [hunk 2]
```diff
@@ -201,6 +201,54 @@ Extra parameters might be available for specific widgets.
 }
 ```
 
+### resourceGraph parameters
+
+- **depth** - defines the maximum distance from original resource to transitively related resource. Defaults to infinity.
```
Comment (grego952) [inline docs/extensibility/resources.md:206]:
```suggestion
- **depth** - defines the maximum distance from the original resource to a transitively related resource. Defaults to infinity.
```

File: docs/extensibility/resources.md [hunk 3]
```diff
@@ -201,6 +201,54 @@ Extra parameters might be available for specific widgets.
...
+- **colorVariant** - optional integer in range 1 to 11 or 'neutral', denoting the SAP color variant of node's border. If not set, the node's border is the same as current text color.
```
Comment (grego952) [inline docs/extensibility/resources.md:207]:
```suggestion
- **colorVariant** - optional integer in range 1 to 11 or 'neutral', denoting the SAP color variant of node's border. If not set, the node's border is the same as the current text color.
```

File: docs/extensibility/resources.md [hunk 4]
```diff
@@ -201,6 +201,54 @@ Extra parameters might be available for specific widgets.
...
+- **networkFlowKind** - optional boolean which determines if the resource should be shown on network graph, Defaults to `false`, which displays the resource on structural graph.
```
Comment (grego952) [inline docs/extensibility/resources.md:208]:
```suggestion
- **networkFlowKind** - optional boolean which determines if the resource should be shown on the network graph, Defaults to `false`, which displays the resource on the structural graph.
```

File: docs/extensibility/resources.md [hunk 5]
```diff
@@ -201,6 +201,54 @@ Extra parameters might be available for specific widgets.
...
+- **networkFlowLevel** - optional integer which sets the horizontal position of resource's node on a network graph.
```
Comment (grego952) [inline docs/extensibility/resources.md:209]:
```suggestion
- **networkFlowLevel** - optional integer which sets the horizontal position of the resource's node on the network graph.
```

File: docs/extensibility/resources.md [hunk 6]
```diff
@@ -201,6 +201,54 @@ Extra parameters might be available for specific widgets.
...
+- **dataSources** - an array of objects in shape:
+  - **source** - a string that must correspond to one of [dataSources](#datasources-section) name. It selects the related resource and the way it should be matched.
```
Comment (grego952) [inline docs/extensibility/resources.md:211]:
```suggestion
  - **source** - a string that must correspond to one of the [dataSources](#datasources-section) name. It selects the related resource and the way it should be matched.
```

File: docs/extensibility/resources.md [hunk 7]
```diff
@@ -254,18 +302,14 @@ Those fields are used to build the related resource URL and filter the received
   - **namespace** - the resource's Namespace name; it defaults to the original resource's Namespace. If set to `null`, cluster-wide resources or resources in all Namespaces are matched.
   - **name** - a specific resource name; leave empty to match all resources of a given type.
 - **ownerLabelSelectorPath** - the path to original object's **selector** type property; for example, `spec.selector.matchLabels` for Deployment, used to select matching Pods.
-- **filter** - [JSONata](https://docs.jsonata.org/overview.html) function enabling the user to write a custom matching logic. It receives a data context of:
+- **filter** - [JSONata](https://docs.jsonata.org/overview.html) function enabling the user to write a custom matching logic. It can use the following variables:
```
Comment (grego952) [inline docs/extensibility/resources.md:305]:
```suggestion
- **filter** - [JSONata](https://docs.jsonata.org/overview.html) function enabling the user to write a custom matching logic. It uses the following variables:
```

File: docs/extensibility/resources.md [hunk 8]
```diff
@@ -254,18 +302,14 @@ Those fields are used to build the related resource URL and filter the received
...
 
-  ```js
-  {
-    data, // related resource
-    resource, // original resource
-  }
-  ```
+  - **item** - the current item of a related kind.
```
Comment (grego952) [inline docs/extensibility/resources.md:307]:
```suggestion
  - **item** - the current item of the related kind.
```

**Review by wchrapka — APPROVED** [review]

File: docs/extensibility/resources.md [hunk 9]
```diff
@@ -201,6 +201,54 @@ Extra parameters might be available for specific widgets.
 }
 ```
 
+### resourceGraph parameters
+
+- **depth** - defines the maximum distance from the original resource to a transitively related resource. Defaults to infinity.
+- **colorVariant** - optional integer in range 1 to 11 or 'neutral', denoting the SAP color variant of node's border. If not set, the node's border is the same as the current text color.
```
Comment (grego952) [inline docs/extensibility/resources.md:207]:
```suggestion
- **colorVariant** - optional integer in range 1 to 11 or 'neutral', denoting the SAP color variant of the node's border. If not set, the node's border is the same as the current text color.
```

**Comment by wchrapka** [comment]
<html>
<body>


/test pre-busola-integration-test-k3s-namespace
--



</body>
</html>

**Review by grego952 — DISMISSED** [review]

**Review by wchrapka — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**